app/controllers/concerns/exception_handler.rb

```diff
@@ -0,0 +1,17 @@
+module ExceptionHandler
+  extend ActiveSupport::Concern
+
+  included do
+    rescue_from ActiveRecord::RecordNotFound do |e|
+      json_response({ error: e.message }, :not_found)
+    end
+
+    rescue_from ActiveRecord::RecordInvalid do |e|
+      json_response({ error: e.message }, :unprocessable_entity)
+    end
+
+    rescue_from ActiveRecord::RecordNotDestroyed do |e|
+      json_response({ errors: e.record.errors }, :unprocessable_entity)
+    end
+  end
+end
```

Comment:
the code provides a convenient way to handle these common exceptions and return meaningful JSON responses